### PR TITLE
Handle eth node disconnect

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix https://github.com/raiden-network/raiden/issues/707

### CODE REVIEW

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -70,6 +70,24 @@ def check_transaction_threw(client, transaction_hash):
     return int(transaction['gas'], 0) == int(receipt['gasUsed'], 0)
 
 
+def retry_if_connnection_error(tries, reconnect_wait_secs, method, *args, **kwargs):
+    tries = 5
+    reconnect_wait_secs = 4
```
Comment (palango) [inline raiden/network/rpc/client.py]: The input parameters `tries` and `reconnect_wait_secs` are overwritten here.

File: raiden/network/rpc/client.py [hunk 2]
```diff
@@ -70,6 +70,24 @@ def check_transaction_threw(client, transaction_hash):
...
+    while tries != 0:
+        try:
+            result = method(*args, **kwargs)
+            break
+        except requests.exceptions.ConnectionError:
+            log.info(
+                'Timeout in eth client connection. Is client offline? Trying'
+                ' again in {} secs. {} tries remaining'.format(reconnect_wait_secs, tries),
+            )
+            gevent.sleep(reconnect_wait_secs)
```
Comment (palango) [inline raiden/network/rpc/client.py]: Should we the reconnect trials non-linear? Often something exponential with a limit is used.

File: raiden/network/rpc/client.py [hunk 3]
```diff
@@ -136,10 +154,15 @@ def _query_nonce():
         tx = Transaction(nonce, gasprice, startgas, to, value, data)
         assert hasattr(client, 'privkey') and client.privkey
         tx.sign(client.privkey)
-        result = client.call(
+
+        result = retry_if_connnection_error(
+            5,
+            4,
```
Comment (palango) [inline raiden/network/rpc/client.py]: Should these params be global settings?

**Comment by palango** [comment]
This needs to be rebased.

**Comment by palango** [comment]
Sorry, I somehow only looked at the first commit.

Questions stay the same: should `tries`and `reconnect_wait_secs` be global settings and should we use some exponential increase for the reconnect timer?

**Comment by LefterisJP** [comment]
@palango This code is rather obsolete by now. We need to handle this at some point but it's possible the logic would change.

**Comment by palango** [comment]
It is worth working on? I've got some spare time.

**Comment by LefterisJP** [comment]
Sure if you want to spend the time to rebase this and finish this you are free to do so. As a collaborator you should be able to rebase and force push to PRs.

**Comment by palango** [comment]
Superseeded by #1093 
Closing.

**Pull request closed without merging**